src/mon/OSDMonitor.h

```diff
@@ -111,10 +112,14 @@ struct failure_info_t {
 
 class OSDMonitor : public PaxosService {
   CephContext *cct;
+
+  ParallelPGMapper mapper;                        ///< for background pg work
+  unique_ptr<OSDMapMapping> mapping;              ///< pg <-> osd mappings
```

Comment:
because they are optional, and because the classes weren't necessarily written with re-use in mind.